### PR TITLE
Improve story appearance on the revisions screen

### LINKED ISSUE #12135: Revisions: display story player on comparison screen
## Feature Description

The default comparison view in WordPress is not very useful because it just compares HTML or JSON blobs with each other:

Screenshot 2022-08-22 at 16 56 53

Screenshot 2022-08-22 at 16 56 38

It would be more useful if we could actually render the stories in a nice way, preferably using [`<amp-story-player>`](https://amp.dev/documentation/components/amp-story-player/) (see `Embed` class)

For this to work we'd need to make sure the revisions could be accessed by the logged-in user so that they can be iframed.

Relevant actions and filters:

* `admin_print_footer_scripts` / `admin_print_footer_scripts-revision` to enqueue custom JS on the revisions page
* `wp_get_revision_ui_diff` - Filters the fields displayed in the post revision diff UI.
  Can be used to render custom `<amp-story-player>` markup for revision

Stretch goal:

* The two players could be synched so that they show the same pages

## Additional Context

* Note that we already filter `_wp_post_revision_fields` in the `Story_Post_Type` class. Perhaps it's worth extracting that to its own new class that handles everything related to revisions.

### PR DESCRIPTION
**Pull request by swissspidy**
## Context

The goal is to improve the appearance of stories on the revisions screen.

## Summary

Uses `<amp-story-player>` on the revisions screen to improve the UX for comparing revisions.

## To-do

- [ ] Add more PHP tests

## User-facing changes

Screenshot 2022-09-12 at 15 30 15

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Enable improved autosaves experiment
2. Create a new story with some content
3. Publish story
4. Make some changes to the story and save it
5. Click on the revisions link in the Document tab
6. Verify that the revisions page shows the embedded before/after stories

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #12135

### CODE REVIEW

**Comment by felipebochehin87** [comment]
https://images.zenhubusercontent.com/235435637/0096f1e8-b777-46e1-bfbd-707fef3e04e0/2022_09_13_21_24_32.mp4

File: includes/Story_Revisions.php [hunk 1]
```diff
@@ -0,0 +1,251 @@
+<?php
+/**
+ * AdmiRevisionsn class.
```
Comment (timarney) [inline includes/Story_Revisions.php]: Typo here

**Comment by timarney** [comment]
> Verify that the revisions page shows the embedded before/after stories

Not sure what's going on but I'm still seeing the diff interface

Screen Shot 2022-09-14 at 7 19 53 AM

**Review by spacedmonkey — APPROVED** [review]
LGTM

**Comment by spacedmonkey** [comment]
One thing I noticed, I got a 0 in the revision history. 

Screenshot 2022-09-14 at 14 45 50

**Comment by swissspidy** [comment]
> Not sure what's going on but I'm still seeing the diff interface

@timarney Did you run `composer install` after checking out?

**Comment by timarney** [comment]
> @timarney Did you run composer install after checking out?

That was the issue

**Review by timarney — APPROVED** [review]

**Comment by swissspidy** [comment]
> One thing I noticed, I got a 0 in the revision history.

Unable to reproduce but let's keep an eye on it

**Pull request merged**